optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/lookup/ClassAndPlanningIdComparator.java

```diff
@@ -51,9 +58,9 @@ public int compare(Object a, Object b) {
             return aClass.getName().compareTo(bClass.getName());
         }
         MemberAccessor aMemberAccessor = decisionCache.computeIfAbsent(aClass,
-                ConfigUtils::findPlanningIdMemberAccessor);
+                clazz -> ConfigUtils.findPlanningIdMemberAccessor(domainAccessType, clazz));
         MemberAccessor bMemberAccessor = decisionCache.computeIfAbsent(bClass,
-                ConfigUtils::findPlanningIdMemberAccessor);
+                clazz -> ConfigUtils.findPlanningIdMemberAccessor(domainAccessType, clazz));
```

Comment:
IMPORTANT
Switching method references to anonymous methods that call it can cause performance regression if used within ConstrainStreams, because `Foo::getBar == Foo::getBar` returns true but `foo -> foo.getBar() == foo -> foo.getBar()` returns false and this impacts node sharing.

Reply:
Method References are not stable
https://stackoverflow.com/a/50748788

It doesn't forbid reuse of the same lambda, but the spec doesn't guarantee that.

Reply:
More details on when exactly it will be shared/unshared in the Oracle JVM (different JVM may differ: https://stackoverflow.com/a/23991339)